File: programmers/etc/level1/나누어_떨어지는_숫자_배열.py
def solution(arr, divisor):
    answer = []
    for a in arr:
        if a%divisor == 0:
            answer.append(a)
    return sorted(answer) if answer else [-1]


arr = [5, 9, 7, 10]
divisor = 5
print(solution(arr, divisor)) # [5, 10]

arr = [2, 36, 1, 3]
divisor = 1
print(solution(arr, divisor)) # [1, 2, 3, 36]

arr = [3, 2, 6]
divisor = 10
print(solution(arr, divisor)) # [-1]

File: Java/programmers/etc/level1/나누어떨어지는숫자배열.java
package CodeTest.Java.programmers.etc.level1;

import java.util.*;

public class 나누어떨어지는숫자배열 {
    public List<Integer> solution(int[] arr, int divisor) {
        List<Integer> answer = new ArrayList<>();
        Arrays.sort(arr);
        
        for (int a : arr) {
            if (a%divisor == 0) answer.add(a);
        }
        
        return answer.isEmpty() ? List.of(-1) : answer;
    }
}
